test/datacollector.UnitTests/DataCollectionManagerTests.cs

```diff
@@ -0,0 +1,293 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.DataCollector.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    using Microsoft.VisualStudio.TestPlatform.DataCollector;
+    using Microsoft.VisualStudio.TestPlatform.DataCollector.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class DataCollectionManagerTests
+    {
+        private DataCollectionManager dataCollectionManager;
+        private string defaultRunSettings = "<?xml version=\"1.0\" encoding=\"utf-16\"?>\r\n<RunSettings>\r\n  <DataCollectionRunSettings>\r\n    <DataCollectors >{0}</DataCollectors>\r\n  </DataCollectionRunSettings>\r\n</RunSettings>";
+        private string dataCollectorSettings;
+        private string dataCollectorSettingsWithWrongUri, dataCollectorSettingsWithoutUri;
+
+        private IMessageSink mockMessageSink;
+
+
+        [TestInitialize]
+        public void Init()
+        {
+            CustomDataCollector.Reset();
+            DummyMessageSink.Reset();
+
+            this.mockMessageSink = new DummyMessageSink();
+            this.dataCollectionManager = new DataCollectionManager(new DataCollectionAttachmentManager(), new DummyMessageSink());
+
+            this.dataCollectorSettings = string.Format("<DataCollector friendlyName=\"CustomDataCollector\" uri=\"my://custom/datacollector\" assemblyQualifiedName=\"Microsoft.VisualStudio.TestPlatform.DataCollector.UnitTests.CustomDataCollector, datacollector.UnitTests, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a\" codebase=\"{0}\" />", typeof(CustomDataCollector).GetTypeInfo().Assembly.Location);
```

Comment:
>ft.VisualStudio.TestPlatform.DataCollector.UnitTests.CustomDataCollector, datacollector.UnitTests, Version=15.0.0.0, [](start = 173, length = 116)

Derive this from  typeof(CustomDataCollector).GetTypeInfo().Assembly.AssemblyName